src/pages/tBTC/Bridge/Minting/InitiateMinting.tsx

```diff
@@ -1,43 +1,168 @@
 import { BridgeProcessCardTitle } from "../components/BridgeProcessCardTitle"
+import MintingTransactionDetails from "../components/MintingTransactionDetails"
+
+type RevealDepositErrorType = {
+  code: number
+  message: string
+}
 
 const InitiateMintingComponent: FC<{
   utxo: BitcoinUtxo
   onPreviousStepClick: (previousStep?: MintingStep) => void
 }> = ({ utxo, onPreviousStepClick }) => {
-  const { openModal } = useModal()
+  const { updateState } = useTbtcState()
+  const threshold = useThreshold()
+  const { closeModal } = useModal()
+  const [depositRevealErrorData, setDepositRevealErrorData] =
+    useState<RevealDepositErrorType>()
+
+  const onSuccessfulDepositReveal = () => {
+    updateState("mintingStep", MintingStep.MintingSuccess)
+    // We don't have success modal for deposit reveal so we just closing the
+    // current TransactionIsPending modal.
+    closeModal()
+  }
+
+  const onFailedDepositReveal = (error: RevealDepositErrorType) => {
+    setDepositRevealErrorData(error)
+    closeModal()
+  }
+
+  const { sendTransaction: revealDeposit } = useRevealDepositTransaction(
+    onSuccessfulDepositReveal,
+    onFailedDepositReveal
+  )
+
+  const depositedAmount = BigNumber.from(utxo.value).toString()
+  const confirmations = threshold.tbtc.minimumNumberOfConfirmationsNeeded(
+    utxo.value
+  )
+  const durationInMinutes = getDurationByNumberOfConfirmations(confirmations)
+  // Round up the minutes to the nearest half-hour
+  const hours = (Math.round(durationInMinutes / 30) * 30) / 60
+
+  const hoursSuffix = hours === 1 ? "hour" : "hours"
+  const confirmationsSuffix =
+    confirmations === 1 ? "confirmation" : "confirmations"
+
+  useEffect(() => {
+    const getEstimatedDepositFees = async () => {
+      const { treasuryFee, optimisticMintFee, amountToMint } =
+        await threshold.tbtc.getEstimatedDepositFees(depositedAmount)
+      updateState("mintingFee", optimisticMintFee)
+      updateState("thresholdNetworkFee", treasuryFee)
+      updateState("tBTCMintAmount", amountToMint)
+    }
+
+    getEstimatedDepositFees()
+  }, [depositedAmount, updateState, threshold])
 
-  const confirmDespotAndMint = async () => {
-    openModal(ModalType.TbtcMintingConfirmation, { utxo: utxo })
+  const initiateMintTransaction = async () => {
+    if (depositRevealErrorData) {
+      setDepositRevealErrorData(undefined)
+      console.log("Revealing deposit failed, trying again...")
+    }
+    await revealDeposit(utxo)
   }
 
   return (
     <Box mx={{ base: 0, lg: 10 }}>
+      {depositRevealErrorData ? (
+        <Toast
+          status="error"
+          title="Error."
+          description={`Code: ${depositRevealErrorData?.code}`}
+        >
+          <Toast.CollapsibleDetails>
+            {depositRevealErrorData?.message}
+          </Toast.CollapsibleDetails>
+        </Toast>
+      ) : null}
       <BridgeProcessCardTitle
-        onPreviousStepClick={onPreviousStepClick}
```

Comment:
good catch, i must have missed it
ref commit: 068fb84f

Reply:
Like I said earlier, it should take the user back to `ProvideData` step, not `MakeDeposit` as it does now.

To do that we should change the code in `MintingFlowRouter`:

https://github.com/threshold-network/bitcoin-on-base/blob/30ad747224de109e3656cf295e7785e6176dff8a/src/pages/tBTC/Bridge/Minting/MintingFlowRouter.tsx#L76-L83

to:

```
    case MintingStep.InitiateMinting: {
      return (
        <InitiateMinting
          utxo={utxo!}
          onPreviousStepClick={() =>
            onPreviousStepClick(MintingStep.ProvideData)
          }
        />
      )
    }
```

This is important, because users who revert from step 3 to step 2 will not be able to return to step 3 without refreshing the page. Moreover, this could mistakenly suggest that they need to send bitcoins to the generated deposit address once more.

Reply:
Ref commit: 8604a7e2